theano/tensor/nnet/nnet.py

```diff
@@ -384,6 +384,7 @@ def c_code(self, node, name, inp, out, sub):
             }
         }
 
+        #pragma omp parallel for schedule(static, 32)
```

Comment:
Hi nouiz,

Thanks very much for your kindly comments.

1. Regarding to the 1st issue you point out:
Even inherit from theano.gof.Op, my test result show that the compiler still respect flag openmp={True, False}.
My .theanorc configuration is as below:
```
[global]
device = cpu
floatX = float32
cxx = g++
mode = FAST_RUN
openmp = True
openmp_elemwise_minsize = 10
profile=False
[gcc]
cxxflags = -fopenmp -O3 -opt-prefetch=2 -funroll-loops
[blas]
ldflags = -lmkl_rt
```

2. Regarding to 2nd issue:
Yes. I have done timing. (BTW, becasue our test beds are maintained, I use PC to test it now. I use BDW (Xeon E5 2699 v4) before.)
Below is test result:
```
HW: Intel(R) Core(TM) i7-5960X CPU @ 3.00GHz
workload: [dl4mt](https://github.com/nyu-dl/dl4mt-tutorial.git)
```
.theanorc configuration:
```
[global]
device = cpu
floatX = float32
cxx = g++
mode = FAST_RUN
openmp = True
openmp_elemwise_minsize = 10
profile=False
[gcc]
cxxflags = -fopenmp -O3 -opt-prefetch=2 -funroll-loops
[blas]
ldflags = -lmkl_rt
```

profile(just need focus on softmax and softmaxGrad OPs):
gcc
```
===============================================================
non-opti
-----------------------------------------------------------
Class
---
<% time> <sum %> <apply time> <time per call> <type> <#call> <#apply> <Class name>
  29.4%    29.4%      82.766s       1.42e-01s     Py     584       6   theano.scan_module.scan_op.Scan
  26.9%    56.3%      75.623s       3.94e-01s     C      192       2   theano.tensor.nnet.nnet.Softmax
  23.4%    79.7%      65.695s       1.46e-02s     C     4512      46   theano.tensor.blas.Dot22
  12.1%    91.8%      34.037s       9.88e-04s     C    34440     348   theano.tensor.elemwise.Elemwise
   2.8%    94.6%       7.796s       7.80e-02s     C      100       1   theano.tensor.nnet.nnet.SoftmaxGrad
   2.7%    97.3%       7.587s       4.79e-03s     C     1584      16   theano.tensor.basic.Alloc
   1.8%    99.1%       5.176s       4.74e-03s     C     1092      11   theano.tensor.elemwise.Sum
   0.4%    99.5%       1.115s       6.28e-04s     C     1776      18   theano.tensor.subtensor.IncSubtensor
   0.2%    99.7%       0.576s       8.40e-05s     C     6856      70   theano.tensor.basic.Reshape
   0.2%    99.9%       0.543s       1.81e-03s     C      300       3   theano.tensor.subtensor.AdvancedIncSubtensor1
   0.0%    99.9%       0.123s       6.17e-04s     C      200       2   theano.tensor.blas.Gemm
   0.0%   100.0%       0.108s       1.88e-04s     C      576       6   theano.tensor.subtensor.AdvancedSubtensor1
   0.0%   100.0%       0.012s       2.07e-06s     C     5680      58   theano.tensor.opt.MakeVector
   0.0%   100.0%       0.009s       1.71e-06s     C     5296      54   theano.tensor.elemwise.DimShuffle
   0.0%   100.0%       0.009s       1.71e-06s     C     5264      54   theano.compile.ops.Shape_i
   0.0%   100.0%       0.009s       4.44e-05s     Py     192       2   theano.tensor.basic.ARange
   0.0%   100.0%       0.008s       2.25e-06s     C     3728      38   theano.tensor.subtensor.Subtensor
   0.0%   100.0%       0.004s       9.47e-07s     C     4380      45   theano.tensor.basic.ScalarFromTensor
   0.0%   100.0%       0.001s       1.97e-06s     C      384       4   theano.tensor.basic.AllocEmpty
   0.0%   100.0%       0.000s       7.93e-07s     C      584       6   theano.compile.ops.Rebroadcast

opti
-----------------------------------------------------------   
   <% time> <sum %> <apply time> <time per call> <type> <#call> <#apply> <Class name>
  35.9%    35.9%      85.374s       1.46e-01s     Py     584       6   theano.scan_module.scan_op.Scan
  27.1%    63.0%      64.548s       1.43e-02s     C     4512      46   theano.tensor.blas.Dot22
  17.7%    80.7%      42.142s       1.22e-03s     C    34440     348   theano.tensor.elemwise.Elemwise
  10.3%    90.9%      24.428s       1.27e-01s     C      192       2   theano.tensor.nnet.nnet.Softmax
   4.2%    95.1%      10.074s       6.36e-03s     C     1584      16   theano.tensor.basic.Alloc
   2.2%    97.4%       5.295s       4.85e-03s     C     1092      11   theano.tensor.elemwise.Sum
   1.6%    98.9%       3.735s       3.73e-02s     C      100       1   theano.tensor.nnet.nnet.SoftmaxGrad
   0.5%    99.4%       1.122s       6.32e-04s     C     1776      18   theano.tensor.subtensor.IncSubtensor
   0.2%    99.7%       0.590s       8.61e-05s     C     6856      70   theano.tensor.basic.Reshape
   0.2%    99.9%       0.533s       1.78e-03s     C      300       3   theano.tensor.subtensor.AdvancedIncSubtensor1
   0.1%    99.9%       0.122s       6.09e-04s     C      200       2   theano.tensor.blas.Gemm
   0.0%   100.0%       0.112s       1.95e-04s     C      576       6   theano.tensor.subtensor.AdvancedSubtensor1
   0.0%   100.0%       0.013s       2.33e-06s     C     5680      58   theano.tensor.opt.MakeVector
   0.0%   100.0%       0.009s       1.79e-06s     C     5296      54   theano.tensor.elemwise.DimShuffle
   0.0%   100.0%       0.009s       1.79e-06s     C     5264      54   theano.compile.ops.Shape_i
   0.0%   100.0%       0.009s       2.51e-06s     C     3728      38   theano.tensor.subtensor.Subtensor
   0.0%   100.0%       0.009s       4.85e-05s     Py     192       2   theano.tensor.basic.ARange
   0.0%   100.0%       0.004s       1.02e-06s     C     4380      45   theano.tensor.basic.ScalarFromTensor
   0.0%   100.0%       0.001s       1.83e-06s     C      384       4   theano.tensor.basic.AllocEmpty
   0.0%   100.0%       0.000s       4.39e-06s     C      100       1   theano.compile.ops.DeepCopyOp
```

Please help check it.

Thanks a lot.
Kevin Yu